include/tudat/astro/system_models/selfShadowing.h

```diff
@@ -0,0 +1,491 @@
+/*    Copyright (c) 2010-2019, Delft University of Technology
+*    All rigths reserved
+*
+*    This file is part of the Tudat. Redistribution and use in source and
+*    binary forms, with or without modification, are permitted exclusively
+*    under the terms of the Modified BSD license. You should have received
+*    a copy of the license with this file. If not, please or visit:
+*    http://tudat.tudelft.nl/LICENSE.
+*/
+#ifndef TUDAT_SELFSHADOWING_H
+#define TUDAT_SELFSHADOWING_H
+
+#include <map>
+#include <iostream>
+#include <algorithm>
+#include <memory>
+
+#include <thread>
+#include "tudat/astro/system_models/geometryUtils.h"
+#include "tudat/astro/system_models/vehicleExteriorPanels.h"
+
+namespace tudat
+{
+
+namespace system_models
+{
+
+inline bool firstDiscrimination( const std::shared_ptr< system_models::VehicleExteriorPanel > panel, 

```

Comment:
Could you move these inline functions to the .cpp? It can save on computation time (and is better practice)